### PR TITLE
Closes #470: Tokens Pilot Feedback + Enhancements

### LINKED ISSUE #470: Token Replacement: Pilot Feedback and Enhancements
Capture and address feedback from current Token Replacement Roll out [Pilot].
Rollout Tokens to All remaining pipelines

### PR DESCRIPTION
**Pull request by ahmadabdalla**
#470 

## Change Log

- Removed Remote Tokens (Key Vault) functionality:
  - Removed logic from Validate/Deploy Actions (Remote Tokens Logic + Reference KV Env. Variable)
  - Removed Helper Scripts + Updated all Scripts impacted (Convert Tokens + Local Module Test)
  - Removed Key Vault Environment Variable **(Platform Key Vault)** Reference from all workflows
  - Removed Platform Key Vault + Platform Core Resource Group from the Dependency Pipeline 
  - Removed Remote Key Vault reference from Settings File
  - Removed Remote Token Key Vault from all Documentations (Getting Started, Parameter Tokens, Testing Usage, etc..)
- Scripts (Backend work)
  - Updated Synopsis with the right information
  - Simplified Logging
  - Updated Scripts Names and the function names to be more generic
  - Updated Scripts Parameters to be more generic

## Testing

[Network: Loadbalancers](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.loadbalancers.yml)
[Network: Privateendpoints](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.privateendpoints.yml)
[Network: Virtualnetworks](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualnetworks.yml)
[Authorization: Roleassignments](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.roleassignments.yml)
[.Platform: Dependencies](https://github.com/Azure/ResourceModules/actions/workflows/platform.dependencies.yml) (Bug in one of the components - DNS)

## Type of Change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: utilities/pipelines/tokensReplacement/Convert-TokensInParameterFile.ps1 [hunk 1]
```diff
@@ -3,8 +3,7 @@
 This Function Helps with Testing A Module Locally
 
 .DESCRIPTION
-This Function Helps with Testing A Module Locally. Use this Function To perform Pester Testing for a Module and then attempting to deploy it. It Also allows you to use your own
-subscription ID, Principal ID, tenant ID and other parameters that need to be tokenized.
+This Function Aggregates all the different token types (Default and Local) and then passes them to the Convert Tokens Script to replace tokens in a parameter file
```
Comment (MrMCake) [inline utilities/pipelines/tokensReplacement/Convert-TokensInParameterFile.ps1:6]: -> `aggregates`

File: utilities/pipelines/tokensReplacement/helper/Convert-TokenInFile.ps1 [hunk 2]
```diff
@@ -5,77 +5,75 @@ Recieves an Input Object of Name/Value and Searches the Current File for the Nam
...
+    foreach ($Path in $FilePath) {
         # Extract Required Content From the Input
         try {
             $File = Get-Content -Path $Path
-            $FileName = Split-Path $Path -Leaf
+            $FileName = Split-Path -Path $Path -Leaf
         } catch {
             throw $PSItem.Exception.Message
-            exit
+            return
```
Comment (MrMCake) [inline utilities/pipelines/tokensReplacement/helper/Convert-TokenInFile.ps1:71]: I think the return is redundant as `throw` will exit the function

File: utilities/pipelines/tokensReplacement/Convert-TokensInParameterFile.ps1 (hunk 1 above)
Comment (ahmadabdalla) [reply utilities/pipelines/tokensReplacement/Convert-TokensInParameterFile.ps1]: updated

File: utilities/pipelines/tokensReplacement/helper/Convert-TokenInFile.ps1 (hunk 2 above)
Comment (ahmadabdalla) [reply utilities/pipelines/tokensReplacement/helper/Convert-TokenInFile.ps1]: good pickup. thank you

**Review by MrMCake — APPROVED** [review]

**Pull request merged**